src/pages/SitePermissionPage.qml

```diff
@@ -0,0 +1,94 @@
+/****************************************************************************
+**
+** Copyright (c) 2020 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+import Sailfish.WebView.Controls 1.0
+
+Page {
+    id: page
+
+    property string title
+    property string url
+
+    property var _permissions: {
+        "geolocation": PermissionManager.Prompt
+    }
+
+    signal requestFinished()
+
+    function getCurrentIndex(type) {
+        switch(_permissions[type]) {
+        case PermissionManager.Allow:
+            return 0 // index for "Allow" menu
+        case PermissionManager.Deny:
+            return 1 // index for "Deny" menu
+        default:
+            return 2 // index for "Always ask" menu
+        }
+    }
+
+    ListModel {
```

Comment:
It looks like you have similar requirements with this model as you have in your #728 PR. Given this, it would seem sensible to create a reusable "site permission model" (or using `PermissionModel` or `PermissionFilterProxyModel` if they can be restricted to a single site).

Reply:
I don't have a watcher for site permissions to follow automatically. I can ask them when I open a page or click on a button. Their logic also differs. in this pr I show all possible settings, even if they are not in the exclusion list. 
I have not found a way to unify these three things:
1. A list of exceptions for a permission type. For example geolocation list, I get it to use PermissionProxyModel. PermissionModel will filter temporary permission. Here we only store permanent permissions
2. Site permission indication.  I should show all permissions for a site, both temporary and permanent. I need to store icons in this model
3. Site permission page. I should show a static model. Also have qstrId() for each item. I showi all the possible permission settings for this site. Even if they are not. But I should match the data of the set permissions with this model

Reply:
For the case of the icon sources and the qstrId()s, I don't think these should be stored in the model, because they can be derived from elements in the model. You can therefore perform the transformation in the delegate.

Reply:
OK. I don't think it will be difficult and will take a long time to change the PermissionModel

Reply:
c ++ model removes a lot of problems with ListModel